index.html

```diff
@@ -296,141 +292,170 @@
+            exceptions[1][exception]["year"] == dateObj.getFullYear()) {
+            day = exceptions[1][exception]["schedule"];
         }
     }
-    var currentSlot;
-
-    // Will not get called if there is no school
-    for (currentSlot = 0; currentSlot < days[0][day][school].length; currentSlot++) {
-        switch(days[0][day][school][currentSlot]["period"]) { // Falling through intended
-            case "0": // 0 Period
-            case "A": // A period
-            case "B": // B period
-            case "C": // C period
-            case "D": // D period
-            case "E": // E period
-            case "F": // F period
-            case "G": // G period
-            case "H": // H period
-                for (var k = 0; k < schedule["classes"].length; k++) {
-                    if (schedule["classes"][k]["period"] == days[0][day][school][currentSlot]["period"]) {
-                        if (schedule["classes"][k]["name"] != "Free Period") {
-                            var teacherName = schedule["classes"][k]["teacher"]; // Teacher's fullname
-                            var teacherLast = teacherName.split(" ")[1]; // Teacher's last name
-                            scheduleObj = { name: schedule["classes"][k]["name"],
-                                teacher: teacherName,
-                                teacherLastName: teacherLast,
-                                room: schedule["classes"][k]["room"],
-                                period: days[0][day][school][currentSlot]["period"],
-                                time: ""}
-
-                        } else {
-                            scheduleObj = {name: "Free Period",
-                                teacher: "",
-                                teacherLastName: "",
-                                room: "",
-                                period: days[0][day][school][currentSlot]["period"],
-                                time: ""};
-                        }
-                    }
-                }
-                break;
+    return day;
+  }
 
-            default:
-              scheduleObj = {
-                name: days[0][day][school][currentSlot]["period"],
-                teacher: "",
-                room: "",
-                period: "X",
-                time: ""
-              };
-              if (scheduleObj.name == "Lunch" ||
-                  scheduleObj.name == "Assembly" ||
-                  scheduleObj.name == "US Community") {
-                scheduleObj.room = "LPC Commons";
-              }
-        }
-        if (scheduleObj.teacher == "N/A") {
-          scheduleObj.teacher = "";
-        }
-        if (scheduleObj.room == "N/A") {
-          scheduleObj.room = "";
+  // Generate basic name, teacher, room, time, and period info for a class
+  function createClassEntry(schedule, school, day, currentSlot, type) {
+    switch (days[0][day][school][currentSlot]["period"]) { // Falling through intended
```

Comment:
need "var scheduleObj" declaration here, probably on the first line of the function